### PR TITLE
Add compliant and noncompliant examples of python/pytorch-sigmoid-before-bceloss@v1.0

### PR DESCRIPTION
**Pull request by ankit-amazon**
*Description of changes:*
* Rules Added: python/pytorch-sigmoid-before-bceloss@v1.0
* Added compliant and non compliant examples

### CODE REVIEW

File: src/python/detectors/pytorch_sigmoid_before_bceloss/pytorch_sigmoid_before_bceloss.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+
+# {fact rule=pytorch-sigmoid-before-bceloss@v1.0 defects=1}
+def pytorch_sigmoid_before_bceloss_noncompliant():
+    import torch
+    import torch.nn as nn
+    # Noncompliant: `Sigmoid` layer followed by `BCELoss` is less robust.
```
Comment (linghuiluo) [inline src/python/detectors/pytorch_sigmoid_before_bceloss/pytorch_sigmoid_before_bceloss.py:8]: is not numerically robust

**Review by grl-wang — APPROVED** [review]
